src/components/AndelaCentres/AndelaCentresComponent.jsx

```diff
@@ -55,77 +63,110 @@ class AndelaCentresComponent extends React.Component {
     this.handleToggleModal();
   };
 
-  handleToggleModal = () => this.setState({ modalOpen: !this.state.modalOpen });
+  handleEditToggleModal = (id) => {
+    this.setState({
+      editModalOpen: !this.state.editModalOpen,
+      locationId: id
+    });
+    this.props.resetMessage();
+  }
+
+  handleEditSubmit = (event) => {
+    event.preventDefault();
+
+    this.props.updateAndelaCentre(this.state.locationId, {
+      centre_name: this.state.centre,
+      country: this.state.country
+    });
+  };
 
   getTotalPages = () => Math.ceil(this.props.locationCount / this.state.limit);
 
   render() {
-    const { isLoading, locationList, error, resetMessage } = this.props;
+    const { isLoading, locationList, error, resetMessage, entity } = this.props;
     const hasLocations = !isEmpty(locationList);
     const showStatus = error;
+    const showAction = entity === 'andela-centres';
     const showNotFound = !isLoading && !hasLocations && !showStatus;
 
     return (
       <NavBarComponent>
-        <div className="assets-list">
-          <PageHeader header="Andela Centres">
-            <div className="header-modal-button">
-              <ModalComponent
-                trigger={
-                  <Button className="add-asset" size="medium">
+        <PageHeader header="Andela Centres">
+          <div className="header-modal-button">
+            <ModalComponent
+              trigger={
+                <Button className="add-asset" size="medium">
                     ADD CENTRE
-                  </Button>
+                </Button>
                 }
-                modalTitle="Add Centre"
-                toggleModal={this.handleToggleModal}
-                modalOpen={this.state.modalOpen}
-              >
-                <CentreModal
-                  handleChange={this.handleChange}
-                  handleSubmit={this.handleSubmit}
-                  countries={this.props.countries}
-                  onSelectCountry={this.onSelectCountry}
-                  country={this.state.country}
-                />
-              </ModalComponent>
-            </div>
-          </PageHeader>
-
-          <div className="assets-list">
-            {showStatus && (
-              <StatusMessageComponent
-                message={error}
-                className="error-status"
-                reset={resetMessage}
+              modalTitle="Add Centre"
+              toggleModal={this.handleToggleModal}
+              modalOpen={this.state.modalOpen}
+            >
+              <CentreModalContainer
+                handleChange={this.handleChange}
+                handleSubmit={this.handleSubmit}
+                onSelectCountry={this.onSelectCountry}
+                country={this.state.country}
               />
-            )}
+            </ModalComponent>
           </div>
+        </PageHeader>
+
+
+        <ModalComponent
```

Comment:
This block is similar to the one in the `<PageHeader>`. Please create a component for it in `AndelaCentres` directory to avoid repeating the code. As you have realized, you had to make the same changes twice, which is an indication that a new component will be more appropriate in this situation.

Note: The following example hasn't been tested. Ensure that everything works before submitting the changes
```jsx
const AndelaCentersModal = props => {
  const {
    showTrigger,
    title, 
    onToggleModal,
    modalOpen,
    onChange,
    onSubmit,
    onSelectCountry,
    country
  } = props;
  const trigger =  (
    <Button className="add-asset" size="medium">
       ADD CENTRE
    </Button>
  );
  return (
    <ModalComponent
      trigger={showTrigger ? trigger : null}
      modalTitle={title}
      toggleModal={onToggleModal}
      modalOpen={modalOpen}
    >
      <CentreModalContainer
         handleChange={onChange}
         handleSubmit={onSubmit}
         onSelectCountry={onSelectCountry}
         country={country}
       />
    </ModalComponent>
  );
}
```

In `AndelaCentresComponent.jsx`:
```jsx
render() {
  ...
  return (
    <NavBarComponent>
      <PageHeader header="Andela Centres">
        <div className="header-modal-button">
          <AndelaCentersModal
            showTrigger
            title="Add Center"
            ...
            country={this.state.country}
          />
        </div>
      </PageHeader>
      ...
    </NavBarComponent>
  );
}